### PR TITLE
Fix: time reg form validation opens all other time regs in modal

### PR DESCRIPTION
**Pull request by Jordybastien**
### What this PR is adding

This PR introduces `phlex-ui` and a modal component for creating and editing time registrations(time_regs).

### Functionality

Modals are opened/triggered using a Turbo Stream and a corresponding route. This initial implementation required the modal to be partially open by default within the partial.

The chosen `phlex-ui` component initially had a limitation when opening modals by default [see submitted issue](https://github.com/PhlexUI/phlex_ui_stimulus/issues/7), luckily a fix was implemented, it was unfortunately placed in the wrong component. This is why you'll see the `custom_dialog_controller.js` controller. Its sole purpose is to ensure the modal is open when the Turbo Stream link is visited.

Another challenge with the current implementation is how `phlex-ui` handles modals. It creates a DOM template that's inserted each time the modal is opened [reference](https://github.com/PhlexUI/phlex_ui_stimulus/blob/main/controllers/dialog_controller.js#L15), This complexity arises when trying to remove/close the modal during a Turbo Stream update/create from the backend using `turbo_stream.remove :modal`. The issue lies with the newly inserted modal node component remaining in the DOM even after removing the turbo_frame_tag wrapper div. This leads to a visually unappealing situation and requires manually calling the close modal button.

### Current introduced Solution

To address this issue, I've opted to create a custom stream action that mimics the functionality of `turbo_stream.remove` but also ensures `phlex-ui` components are properly removed/closed from the DOM.

### How to test?
Create and edit time_reg

### What's missing?
In the modal form you will notice that `text_input` and `buttons` are not using `phlex-ui` components. That's because I think these components could be defined/configured within one single place `components` folder and every other place using them just call them with minimal effort/configuration possible. I would like to do that in a separate PR, and avoid this one to continue growing

### CODE REVIEW

**Comment by str1fe** [comment]
Great idea @Jordybastien. I would like to propose one change though :smiley: 

I look at this project kind of like a test bench where we can try some new tools, that we have not used before. That's why we're using minitest and fixtures.

When it comes to the view_component gem, i have used it multiple times, but i've been keeping an eye on a project called Phlex, which i think is interesting (https://github.com/phlex-ruby/phlex-rails). Would you be willing to try that instead of view_components?

**Comment by Jordybastien** [comment]
> Great idea @Jordybastien. I would like to propose one change though 😃
> 
> I look at this project kind of like a test bench where we can try some new tools, that we have not used before. That's why we're using minitest and fixtures.
> 
> When it comes to the view_component gem, i have used it multiple times, but i've been keeping an eye on a project called Phlex, which i think is interesting (https://github.com/phlex-ruby/phlex-rails). Would you be willing to try that instead of view_components?

Nice, That's interesting.
I'd love to try that, I'll use it instead of `view_components`

**Comment by str1fe** [comment]
Awesome!

File: app/views/layouts/application_layout.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+				stylesheet_link_tag "application", data_turbo_track: "reload"
+				javascript_importmap_tags
+			end
+
+			body do
+				main(&block)
+			end
+		end
+	end
+end
```
Comment (str1fe) [inline app/views/layouts/application_layout.rb:24]: I'm excited about this, @Jordybastien :raised_hands: 

Indentation is a bit crazy though :laughing:

File: app/views/time_regs/update.turbo_stream.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<% if @time_reg.errors.any? %>
+  <%= turbo_stream.replace :modal, partial: "time_regs/form", locals: { time_reg: @time_reg, chosen_date: @chosen_date, projects: @projects, title: "Edit time entry", assigned_tasks: @assigned_tasks } %>
+<% else %>
+  <turbo-stream action="remove_modal" target="modal"></turbo-stream>
+  <%= turbo_stream.replace dom_id(@time_reg), partial: "time_regs/time_reg", locals: { time_reg: @time_reg } %>
+<% end %>
```
Comment (str1fe) [inline app/views/time_regs/update.turbo_stream.erb:6]: Updating only each individual item is very performant, but i am curious if we could solve this using the new turbo 8 with dom morphing. I think that could potentially save us some work. Wdyt? It should be done in a seperate PR though, after this one is merged

**Review by str1fe — APPROVED** [review]
:raised_hands:

File: app/views/time_regs/update.turbo_stream.erb (hunk 2 above)
Comment (Jordybastien) [reply app/views/time_regs/update.turbo_stream.erb:6]: Nice, let me check that out. I didn't know about it

**Pull request merged**